util/statvar_dcid_generator.py

```diff
@@ -0,0 +1,193 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""A utility to generate dcid for statistical variables."""
+
+import copy
+import re
+
+#Constants
+_QUANTITY_REGEX = r"\[(?P<value>-|-?\d+(\.\d+)?) (?P<quantity>[A-Za-z]+)\]"
+_QUANTITY_RANGE_REGEX = (r"\[(?P<lower_limit>-|-?\d+(\.\d+)?) "
+                         r"(?P<upper_limit>-|-?\d+(\.\d+)?) "
+                         r"(?P<quantity>[A-Za-z]+)\]")
+
+# This is a lookup to ensure "Mean" and "Median" are used in the dcid when
+# statType is meanValue or medianValue"
+_stat_type_lookup = {"meanValue": "Mean", "medianValue": "Median"}
+
+# These are the default properties ignored during dcid generation
+_ignore_props = {"unit", "Node", "memberOf", "typeOf"}
+
+
+def capitalize_word(word):
+    """Capitalizes the first character of a string.
+
+    This function differs from the builtin function str.capitalize() in that it
+    changes only the case of the first character and ignores the case of other
+    characters.
+    """
+    if word:
+        return word[0].upper() + word[1:]
+    return None
+
+
+def generate_quantity_range_name(match_dict):
+    """Generate a name for a quantity range.
+
+    Args:
+        match_dict: A dictionary containing quantity range regex groups.
+          Expected syntax of match_dict is
+          {
+            "lower_limit": <value>,
+            "upper_limit": <value>,
+            "quantity": <value>
+          }
+
+    Returns:
+        A string representing the quantity range name to be used in the dcid.
+    """
+    lower_limit = match_dict["lower_limit"]
```

Comment:
if the match_dict doesn't contain "lower_limit" or "upper_limit" or "quantity"
 this function will generate an exception and terminate. Is that expected?

May be better to return 'None' for such cases and let the caller handle errors appropriately.